gym/vector/utils/info_strategies.py

```diff
@@ -0,0 +1,99 @@
+"""
+Strategies for processing the info dictionary of VecEnvs
+"""
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from gym.error import InvalidInfoFormat
+
+
+class VecEnvInfoStrategy(ABC):
+    """Interface for implementing a info
+    processing strategy for vectorized environments
+    """
+
+    @abstractmethod
+    def __init__(self, num_envs: int):
+        ...
+
+    @abstractmethod
+    def add_info(self, info: dict, env_num: int):
+        """Get the info dict from the
+        environment and process with the defined strategy
+        """
+
+    @abstractmethod
+    def get_info(self):
+        """Return the info for the
+        vectorized env
+        """
+
+
+class ClassicVecEnvInfoStrategy(VecEnvInfoStrategy):
+    """Process the info dictionary of an environment
+    so that the vectorized info is returned in the form of a list of dictionaries.
+
+    Example with 3 environments:
+        [{}, {}, {}]
+    """
+
+    def __init__(self, num_envs: int):
+        self.num_envs = num_envs
+        self.info = []
+
+    def add_info(self, info: dict, env_num: int):
+        self.info.append(info)
+
+    def get_info(self) -> list:
+        return self.info
+
+
+class BraxVecEnvInfoStrategy(VecEnvInfoStrategy):
+    """Process the info dictionary of an environment
+    so that the vectorized info is returned in the form of a single dictionary.
+    Keys of the dictionary represents the `info` key; Values are lists
+    in which each index correspond to an environment. If the environment
+    at index `i` does not have a value for `info` then it is set to `None`
+
+    This strategy matches Brax library info's output structure.
+
+    Example with 3 environments in which only the last has the `terminal_observation` info:
+        {
+            "terminal_observation": [
+                None,
+                None,
+                array([0.13,  1.58 , -0.22, -2.56])
+            ]
+        }
+    """
+
+    def __init__(self, num_envs: int):
+        self.num_envs = num_envs
+        self.info = {}
+
+    def add_info(self, info: dict, env_num: int):
+        for k in info.keys():
+            info_array = self.info.get(k, [None for _ in range(self.num_envs)])
```

Comment:
For optimisation, should this not be a numpy array? This adds some complications for the dtype of the array, but I think we should be able to check if the data is a numpy type and use that as the dtype otherise dtype=object

Reply:
It can be an idea; how do you think it should be initialized? np.nan for every env?

Reply:
I dont think we should use `np.nan` unless necessary. To me it is a special value that shouldn't be used unless it is an actual `nan`
Personally, I was originally thinking `np.zeros` as the dtype should automatically know the `zero` dtype. However for bool, then this get a bit complex as it is quite reasonable to get `False` as info data. We could have a special case just for bools. 
What are your ideas?

Reply:
I've thought about it; this is actually quite inefficient
```python
[None for _ in range(self.num_envs)]
```
and should be at least changed to:
```python
[None] * self.num_envs
```
Which has a creation time comparable to a np.array.
However, I also expect that users want to do some kind of processing on the info and will probably convert the list themselves to a numpy array. With this in mind, I think we are moving towards this approach:
https://github.com/openai/gym/issues/2657#issuecomment-1055597901
We can leverage the new modularity of the `info` and add this as a new strategy with np.arrays initialized as np.zeros (with proper management of the bool case). 
I'm just thinking at the benefits of keeping them both at this point and the only benefit I see is that `None` is clearer than 0's or False (https://github.com/openai/gym/issues/2657#issuecomment-1104474552)